storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -0,0 +1,570 @@
+        }
+    }
+
+    /**
+     * Validates a Integer.
+     */
+    public static class IntegerValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            validateInteger(name, o);
+        }
+
+        public void validateInteger(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            final long i;
+            if (o instanceof Number &&
+                    (i = ((Number) o).longValue()) == ((Number) o).doubleValue()) {
+                if (i <= Integer.MAX_VALUE && i >= Integer.MIN_VALUE) {
+                    return;
+                }
+            }
+            throw new IllegalArgumentException("Field " + name + " must be an Integer within type range.");
+        }
+    }
+
+    /**
+     * Validates a map of Strings to a map of Strings to a list.
+     * {str -> {str -> [str,str]}
+     */
+    public static class ImpersonationAclValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            if (o == null) {
+                return;
+            }
+            ConfigValidationUtils.NestableFieldValidator validator = ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                    ConfigValidationUtils.mapFv(ConfigValidationUtils.fv(String.class, false),
+                            ConfigValidationUtils.listFv(String.class, false), false), true);
+            validator.validateField(name, o);
+        }
+    }
+
+    /**
+     * validates a list of has no duplicates
+     */
+    public static class NoDuplicateInListValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object field) {
+            if (field == null) {
+                return;
+            }
+            //check if iterable
+            SimpleTypeValidator.validateField(name, Iterable.class, field);
+            HashSet<Object> objectSet = new HashSet<Object>();
+            for (Object o : (Iterable) field) {
+                if (objectSet.contains(o)) {
+                    throw new IllegalArgumentException(name + " should contain no duplicate elements. Duplicated element: " + o);
+                }
+                objectSet.add(o);
+            }
+        }
+    }
+
+    /**
+     * Validates a String or a list of Strings
+     */
+    public static class StringOrStringListValidator extends Validator {
+
+        private ConfigValidationUtils.FieldValidator fv = ConfigValidationUtils.listFv(String.class, false);
+
+        @Override
+        public void validateField(String name, Object o) {
+
+            if (o == null) {
+                return;
+            }
+            if (o instanceof String) {
+                return;
+            }
+            //check if iterable
+            try {
+                SimpleTypeValidator.validateField(name, Iterable.class, o);
+            } catch (Exception ex) {
```

Comment:
that is not suppose to be in there.  Not sure how the try catch statement got in there to begin with.  Will remove